src/main/java/org/inferred/freebuilder/processor/Analyser.java

```diff
@@ -528,21 +519,28 @@ public Boolean visitDeclared(DeclaredType t, Void p) {
       };
 
   /**
-   * Verifies the given method is an abstract getter following the JavaBean convention. Any
+   * Verifies {@code method} is an abstract getter following the JavaBean convention. Any
    * deviations will be logged as an error.
+   *
+   * <p>We deviate slightly from the JavaBean convention by insisting that there must be a
+   * non-lowercase character immediately following the get/is prefix; this prevents ugly cases like
+   * 'get()' or 'getter()'.
+   *
+   * @return a {@link Matcher} with the getter prefix in group 1 and the property name suffix
+   *     in group 2, or {@code null} if {@code method} is not a valid abstract getter method
    */
-  private boolean isValidProperty(TypeElement valueType, ExecutableElement method) {
+  private MatchResult getterNameMatchResult(TypeElement valueType, ExecutableElement method) {
```

Comment:
That wouldn't be consistent with the rest of the code. Also, I think static analysers only need it across compilation units.